### PR TITLE
improve several tests in testament

### PR DESCRIPTION
**Pull request by timotheecour**

### CODE REVIEW

File: tests/iter/tyieldintry.nim [hunk 1]
```diff
@@ -1,17 +1,15 @@
 discard """
-targets: "c"
-output: "ok"
+  targets: "c cpp"
 """
-var closureIterResult = newSeq[int]()
 
-# XXX Investigate why this fails now for 'nim cpp'
```
Comment (Varriount) [inline tests/iter/tyieldintry.nim:7]: Does this test still fail for `nim cpp`?

Comment (timotheecour) [reply tests/iter/tyieldintry.nim:7]: targets: "c cpp"

Comment (Varriount) [reply tests/iter/tyieldintry.nim:7]: I don't understand. Does it still fail when compiled and run using the C++ backend?

Comment (timotheecour) [reply tests/iter/tyieldintry.nim:7]: see the diff, it adds `targets: "c cpp"` so, cpp is now tested

**Pull request merged**